src/json/print.js

```diff
@@ -9,23 +9,16 @@ export default (obj, dataPath, options = {}) => annotate => {
     annotationColor = 'cyan',
     gutterChar = '│',
     annotationGutterChar = '〉',
+    indent = 2,
   } = options;
 
-  const jsonString = JSON.stringify(obj, null, 2);
-  const lines = jsonString.split('\n');
-  const gutterWidth = lines.length.toString().length;
-  const annotationGutter = `${''.padStart(gutterWidth)}${annotationGutterChar}`;
-
-  // console.log(JSON.stringify(parse(jsonString), null, 2));
-  // TODO: Handle json pointer escape notation and better error handling
   try {
-    const { meta: { line, column } } = dataPath
-      .split('/')
-      .slice(1)
-      .reduce(
-        (obj, pointer) => obj.value[pointer] || obj.value.value[pointer],
-        parse(jsonString)
-      );
+    const { line, column } = findErrorPosition(obj, dataPath, indent);
+
+    const jsonString = JSON.stringify(obj, null, indent);
```

Comment:
Thanks for extracting it into a different file. But can we pass the `jsonString` as param in `findErrorPosition`. It will save us from doing `JSON.stringify` twice. It's a costly function.

Reply:
fixed e262593